### PR TITLE
Introduce `cider-ignored-error-phases`

### PR DESCRIPTION
**Pull request by vemv**
Commits:

* Use cider-nrepl 0.36.0
  * Supports this PR, and also [broadens](https://github.com/clojure-emacs/haystack/pull/11) the semantics of the `tooling` filter:

image

* Introduce `cider-ignored-error-phases`
  * Fixes https://github.com/clojure-emacs/cider/issues/3418
* `cider-auto-jump-to-error`: also recognise macroexpansion errors
  * Mostly unrelated, but also very small.

### QA

This PR's main focus is easy to QA:

* `(setq cider-show-error-buffer 'always)`
* enter an invalid token like `::::a` in the REPL
  * no stacktrace will show up, a concise message will be printed at the repl
* enter an invalid macroexpansion like `(let [1])` in the REPL
  * no stacktrace will show up, a concise message will be printed at the repl
* enter an expr that will cause a runtime exception like `(/ 2 0)`
  * the `*cider-error*` buffer will show up
  * if the `tooling` filter is on, notably few stack frames will be visible
  * the usual output will also be visible at the REPL output.

Cheers - V

### CODE REVIEW

**Comment by vemv** [comment]
@yuhan0 : feel free to review this one, I reckon it might interest you

File: cider-eval.el [hunk 1]
```diff
@@ -478,32 +485,63 @@ op/situation that originated this error."
     (let ((error-buffer (cider-new-error-buffer #'cider-stacktrace-mode error-types)))
       (cider-stacktrace-render error-buffer (reverse causes) error-types))))
 
-(defun cider--handle-stacktrace-response (response causes)
-  "Handle stacktrace op RESPONSE, aggregating the result into CAUSES.
+(defcustom cider-ignored-error-phases '("read-source"
```
Comment (bbatsov) [inline cider-eval.el:488]: I think this name is very confusing, as it doesn't indicate any relationship with the stacktrace display functionality. I also have issue with the term "ignored", as I don't think it reflects what we are really doing here.

File: cider-eval.el [hunk 2]
```diff
@@ -478,32 +485,63 @@ op/situation that originated this error."
...
+                                        "macroexpansion"
+                                        "compile-syntax-check"
+                                        "compilation"
+                                        ;; "execution" is certainly not to be included here.
+                                        ;; "read-eval-result" and "print-eval-result" are not to be included here,
+                                        ;; because they mean that the code has been successfully executed.
+                                        )
+  "Clojure error phases which will not trigger a UI to become visible.
+
+Those UIs include, at the moment:
```
Comment (bbatsov) [inline cider-eval.el:499]: Given there's just one UI, I'm guessing you can simplify the wording here. :-)

File: cider-eval.el (hunk 1 above)
Comment (bbatsov) [reply cider-eval.el:488]: Perhaps just name this `cider-clojure-compilation-error-phases` or something like this? I doubt anyone would be changing this one much.

**Comment by bbatsov** [comment]
You also need to update the "dealing with errors" docs.

**Comment by vemv** [comment]
Thanks for the review!

I've amended the last commit applying all feedback.

**Comment by vemv** [comment]
Amended again 

The only change:

> In the docs I call defcustom "configuration option, as I think it's more newcomer friendly.

**Comment by bbatsov** [comment]
Thanks!
